lib/hybrid-sdk/client/socket.ts

```diff
@@ -176,63 +176,70 @@ export const createWebSocket = (
-    timeoutHandler = async () => {
-      clearTimeout(websocket.timeoutHandlerId);
+  if (getAuthConfig().accessToken && clientOpts.config.UNIVERSAL_BROKER_GA) {
+    // The access token is not sent when making requests over the websocket, so we must re-validate
+    // the token periodically to keep the connection alive. Connections that do not re-validate the
+    // access token will be marked as stale and closed by the server.
+    const timeoutHandler = async () => {
+      const commonLogFields = {
+        connection: maskToken(identifyingMetadata.identifier),
+        role: identifyingMetadata.role,
+      };
+
+      websocket.transport.extraHeaders = {
+        Authorization: getAuthConfig().accessToken.authHeader,
+        'x-snyk-broker-client-id': identifyingMetadata.clientId,
+        'x-snyk-broker-client-role': identifyingMetadata.role,
+        'x-broker-client-version': version,
+      };
 
-      if (clientOpts.config.UNIVERSAL_BROKER_GA) {
-        websocket.transport.extraHeaders = {
-          Authorization: getAuthConfig().accessToken.authHeader,
-          'x-snyk-broker-client-id': identifyingMetadata.clientId,
-          'x-snyk-broker-client-role': identifyingMetadata.role,
-          'x-broker-client-version': version,
-        };
+      logger.debug(commonLogFields, 'Renewing auth.');
+      const renewResponse = await renewBrokerServerConnection(
+        {
+          connectionIdentifier: identifyingMetadata.identifier!,
+          brokerClientId: identifyingMetadata.clientId,
+          authorization: getAuthConfig().accessToken.authHeader,
+          role: identifyingMetadata.role,
+          serverId: serverId,
+        },
+        clientOpts.config,
+      );
 
-        logger.debug(
-          {
-            connection: maskToken(identifyingMetadata.identifier),
-            role: identifyingMetadata.role,
-          },
-          'Renewing auth.',
-        );
-        const renewResponse = await renewBrokerServerConnection(
-          {
-            connectionIdentifier: identifyingMetadata.identifier!,
-            brokerClientId: identifyingMetadata.clientId,
-            authorization: getAuthConfig().accessToken.authHeader,
-            role: identifyingMetadata.role,
-            serverId: serverId,
-          },
-          clientOpts.config,
-        );
-        if (renewResponse.statusCode != 201) {
-          logger.debug(
+      const statusClass = Math.floor((renewResponse.statusCode ?? 0) / 100);
+      switch (statusClass) {
+        case 2: // 2XX - success
+          logger.debug(commonLogFields, 'Auth renewed.');
+          break;
+        case 4: // 4XX - client error - unrecoverable
+          logger.fatal(
             {
-              connection: identifyingMetadata.identifier,
-              role: identifyingMetadata.role,
+              ...commonLogFields,
               responseCode: renewResponse.statusCode,
             },
-            'Failed to renew connection.',
```

Comment:
Renewal loop quietly stops, auth becomes stale and connection is terminated by server